src/main/java/solaire/data/task/Deadline.java

```diff
@@ -0,0 +1,63 @@
+package solaire.data.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import solaire.data.exception.SolaireException;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * Creates a task with a specified deadline.
+     *
+     * @param taskName description of the task.
+     * @param ddl      deadline as a String.
+     * @throws SolaireException if ddl does not follow the LocalDate format specified.
+     */
+    public Deadline(String taskName, String ddl) throws SolaireException {
+        super(taskName);
+        this.by = parseDeadline(ddl);
+    }
+
+    private LocalDate parseDeadline(String ddl) throws SolaireException {
+        try {
+            DateTimeFormatter f = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            return LocalDate.parse(ddl, f);
+        } catch (DateTimeParseException e) {
+            throw new SolaireException("Deadline must be in the format: yyyy-mm-dd");
+        }
+    }
+
+    public LocalDate getDeadline() {
+        return this.by;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
It might be easier to use a formatter to convert the date to a format that you want and then print it out, it would reduce the need for you to extract the month, day and year individually here!